acc_test/resource_kafka_topic_test.go

```diff
@@ -0,0 +1,211 @@
+package test
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/instaclustr/terraform-provider-instaclustr/instaclustr"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strconv"
+	"testing"
+)
+
+func TestKafkaTopicResource(t *testing.T) {
+	testProviders := map[string]terraform.ResourceProvider{
+		"instaclustr": instaclustr.Provider(),
+	}
+
+	configBytes1, _ := ioutil.ReadFile("data/kafka_topic_create_cluster.tf")
+	configBytes2, _ := ioutil.ReadFile("data/kafka_topic_create_topic.tf")
+	configBytes3, _ := ioutil.ReadFile("data/kafka_topic_topic_list.tf")
+	configBytes4, _ := ioutil.ReadFile("data/invalid_kafka_topic_create_duplicate.tf")
+	configBytes5, _ := ioutil.ReadFile("data/kafka_topic_config_update.tf")
+
+	username := os.Getenv("IC_USERNAME")
+	apiKey := os.Getenv("IC_API_KEY")
+	hostname := getOptionalEnv("IC_API_URL", instaclustr.DefaultApiHostname)
+
+	topic1 := "test1"
+	topic2 := "test2"
+	kafkaNodeSize := "KFK-DEV-t4g.medium-80"
+	zookeeperNodeSize := "KDZ-DEV-t4g.small-30"
+	kafkaVersion := "apache-kafka:3.0.0"
+
+	createClusterConfig := fmt.Sprintf(string(configBytes1), username, apiKey, hostname, kafkaNodeSize, kafkaVersion, zookeeperNodeSize)
+	createKafkaTopicConfig := fmt.Sprintf(string(configBytes2), username, apiKey, hostname, kafkaNodeSize, kafkaVersion, zookeeperNodeSize,
+		topic1,
+		topic2)
+	createKafkaTopicListConfig := fmt.Sprintf(string(configBytes3), username, apiKey, hostname, kafkaNodeSize, kafkaVersion, zookeeperNodeSize,
+		topic1,
+		topic2)
+	invalidKafkaTopicCreateConfigDuplicate := fmt.Sprintf(string(configBytes4), username, apiKey, hostname, kafkaNodeSize, kafkaVersion, zookeeperNodeSize,
+		topic1,
+		topic2,
+		topic1)
+	updateKafkaTopicConfig := fmt.Sprintf(string(configBytes5), username, apiKey, hostname, kafkaNodeSize, kafkaVersion, zookeeperNodeSize,
+		topic1,
+		topic2)
+
+	resource.Test(t, resource.TestCase{
```

Comment:
The acc_tests are really slow and flakey unfortunately (particularly tests that wait for clusters to reach running state). So I'm keen to make sure if we add new tests that they're the absolutely most efficient versions of the tests that can be. Is there any way to minimise the number of clusters we have to wait to reach running?

Reply:
For these tests, I am afraid that they need to have the cluster in RUNNING, because the API calls need cluster to be in RUNNING state.

Reply:
So is this one cluster or five?

Reply:
Only one cluster, one cluster for all the tests.

Reply:
OK Great. We already have acc tests that create Kafka clusters and wait for them to reach running. Are there opportunities to combine? (To reduce the time we spend waiting and the $$ we spend on creating)

Reply:
I think for every acc test, it will destroy the cluster after all steps are completed. Unless I combine resource_kafka_topic_test.go and resource_kafka_user_test.go together. But I don't know whether this is a good idea, because these two tests are testing different things.

Reply:
From a cleanliness of testing point of view I agree. But in this case where we're doing long running operations that cost real time and money, I think it's worth considering if the cluster is really part of your test, or whether it is simply a platform for running your tests of users and topics on. In which case, Why should we create more platforms than we need to?

Reply:
Okay, that sounds reasonable. Then I will combine both users and topics into a more general test file resource_kafka_test.go. Considering we're going to add ACL management soon, I think it worths combining them into one to save time and money as only one cluster is needed for all these operations.

Reply:
Acceptance tests of Kafka user and topic management are combined together.